### PR TITLE
Malf AI overload blacklists

### LINKED ISSUE #83223: Malf AI's Machine Overload can delete Escape Shuttle Consoles, despite them being flagged as INDESTRUCTIBLE
## Issue Summary

As it says in the title

## Reproduction:
Step 1: Be a malfunctioning AI and get the computer explosion power.
Step 2: Go look at the escape shuttle console.
Step 3: Use your powers to explode the console.
Step 4: Observe as it totally works

Who knows, maybe the AI can delete more things than just this console given the only unique feature of the console is the flag.

### PR DESCRIPTION
**Pull request by grungussuss**
## About The Pull Request
adds more machines to the malf AI overload ability blacklist: shuttle consoles and gateway console
closes https://github.com/tgstation/tgstation/issues/83223
## Why It's Good For The Game
bug fix
## Changelog
:cl:
fix: fixed malf AI being able to overload shuttle consoles and the gateway control console
/:cl:

### CODE REVIEW

**Comment by Helg2** [comment]
Shouldn't it just automatically return if the machine is indestructible?

**Comment by Sadboysuss** [comment]
> Shouldn't it just automatically return if the machine is indestructible?

well apparently it didn't

File: code/modules/antagonists/malf_ai/malf_ai_modules.dm [hunk 1]
```diff
@@ -47,6 +47,20 @@ GLOBAL_LIST_INIT(blacklisted_malf_machines, typecacheof(list(
...
+		/obj/machinery/computer/shuttle/ferry,
+		/obj/machinery/computer/shuttle,
+		/obj/machinery/computer/emergency_shuttle,
+		/obj/machinery/computer/shuttle/caravan/syndicate1,
+		/obj/machinery/computer/shuttle/caravan/syndicate2,
+		/obj/machinery/computer/shuttle/caravan/syndicate3,
+		/obj/machinery/computer/shuttle/hunter,
+		/obj/machinery/computer/shuttle/pirate/drop_pod,
+		/obj/machinery/computer/shuttle/ferry/request,
+		/obj/machinery/computer/shuttle/pod,
```
Comment (Ghommie) [inline code/modules/antagonists/malf_ai/malf_ai_modules.dm:60]:
```suggestion
		/obj/machinery/computer/shuttle,
		/obj/machinery/computer/emergency_shuttle,
```
You don't need to specify all these computer shuttle subtypes; it's a typecache and the list will be procedurally populated on runtime.

**Review by ShizCalev — CHANGES_REQUESTED** [review]
> > Shouldn't it just automatically return if the machine is indestructible?
> 
> well apparently it didn't

it didn't because the abilities DON'T check for indestructible.

https://github.com/tgstation/tgstation/blob/e12e3dfca19130ef604595de7020b568fefdc1f4/code/modules/antagonists/malf_ai/malf_ai_modules.dm#L533-L555
https://github.com/tgstation/tgstation/blob/e12e3dfca19130ef604595de7020b568fefdc1f4/code/modules/antagonists/malf_ai/malf_ai_modules.dm#L445-L467

just need to change the typecache check in those two procs to check for the flag
```

	if((clicked_machine.resistance_flags & INDESTRUCTIBLE) || is_type_in_typecache(clicked_machine, GLOB.blacklisted_malf_machines))
```

**Comment by ShizCalev** [comment]
just need to change those two procs to use this check instead
```

	if((clicked_machine.resistance_flags & INDESTRUCTIBLE) || is_type_in_typecache(clicked_machine, GLOB.blacklisted_malf_machines))
 		to_chat(caller, span_warning("You cannot overload that device!")) 
 		return FALSE 
```

File: code/modules/antagonists/malf_ai/malf_ai_modules.dm [hunk 2]
```diff
@@ -47,6 +47,11 @@ GLOBAL_LIST_INIT(blacklisted_malf_machines, typecacheof(list(
 		/obj/machinery/hypertorus/corner,
 		/obj/machinery/atmospherics/components/binary/valve,
 		/obj/machinery/portable_atmospherics/canister,
+		/obj/machinery/computer/shuttle,
+		/obj/machinery/computer/emergency_shuttle,
+		/obj/machinery/computer/gateway_control,
+
+
```
Comment (ZephyrTFA) [inline code/modules/antagonists/malf_ai/malf_ai_modules.dm:54]: drop these empty lines

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]

**Review by ZephyrTFA — APPROVED** [review]

**Pull request merged**